src/robot/libraries/BuiltIn.py

```diff
@@ -1428,6 +1428,21 @@ def set_variable(self, *values):
         else:
             return list(values)
 
+    @run_keyword_variant(resolve=0)
+    def set_local_variable(self, name, *values):
+        """ Makes a variable available everywhere within the local scope.
+
+        Variables set with this keyword are available within the
+        local scope of the currently executed test case or in the local scope
+        of the keyword in which they are defined. For example, if you set a
+        variable in a user keyword, it is available only in that keyword. Other
+        test cases or keywords will not see variables set with this keyword.
+        """
+        name = self._get_var_name(name)
+        value = self._get_var_value(name, values)
+        self._variables.set_local_variable(name, value)
+        self._log_set_variable(name, value)
```

Comment:
At least one of the acceptance tests should check that the message is logged correctly. That can be done on the `atest/robot` side by using the `Check Log Message` keyword.